File: 234-palindrome-linked-list/234-palindrome-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    bool isPalindrome(ListNode* head) {
        vector<int> nodes;
        ListNode* curr = head;
        
        while(curr) {
            nodes.push_back(curr->val);
            curr = curr->next;
        } 
        
        for (int i = 0, j = nodes.size() - 1; i < j; i++, j--) {
            if (nodes[i] != nodes[j])
                return false;
        }
        
        return true;
    }
};

File: 234-palindrome-linked-list/234-palindrome-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        nodes = deque()
        curr = head
        length = 0
        
        while (curr):
            nodes.append(curr)
            curr = curr.next
            length += 1
            
        for i in range((int) (length / 2)):
            if (nodes.pop().val != nodes.popleft().val):
                return False
        
        return True
            